browser/brave_news/direct_feed_fetcher_delegate_impl.cc

```diff
@@ -15,17 +16,21 @@ DirectFeedFetcherDelegateImpl::DirectFeedFetcherDelegateImpl(
     HostContentSettingsMap* host_content_settings_map)
     : host_content_settings_map_(host_content_settings_map),
       https_upgrade_exceptions_service_(
-          g_brave_browser_process->https_upgrade_exceptions_service()) {}
+          g_brave_browser_process->https_upgrade_exceptions_service()) {
+  CHECK(host_content_settings_map_);
+  CHECK(https_upgrade_exceptions_service_);
```

Comment:
Is this going to crash the browser if `chrome://flags/#https-by-default` is disabled?

Reply:
nope, just tested